src/Raven.Server/Documents/Indexes/Static/DynamicArray.cs

```diff
@@ -105,6 +105,11 @@ public override bool TryConvert(ConvertBinder binder, out object result)
             return base.TryConvert(binder, out result);
         }
 
+        public dynamic this[int i]
```

Comment:
That has a _high_ cost associated with it.
We have to generate the value each time.
In that case, we are better off materializing the value and then using it.

Reply:
`ElementAt` will read the enumerable, which will rebuild it each time.
We can assume that if the impl is `List<T>`, it is optimized, so we should do an explicit check.